#VRML_SIM R2025a utf8
PROTO Chilaquil [
  field SFVec3f    translation      -0.3 -0.08 0.011  # Posición inicial
  field SFRotation rotation         0 0 1 0           # Rotación inicial
  field SFString   name            "Chilaquil"        # Nombre del robot
  field SFString   controller      "dodge_push_controller"  # Controlador
]
{
  Robot {
    translation IS translation
    rotation IS rotation
    name IS name
    controller IS controller
    children [
      Pose {
        children [
          DEF Body Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
              roughness 1
              metalness 0
            }
            geometry Box {
              size 0.02 0.02 0.02
            }
          }
        ]
      }
      HingeJoint {
        jointParameters HingeJointParameters {
          position 5.52165243934198
          axis 0 1 0
          anchor 0.0001 0.0112 -0.00055
        }
        device [
          RotationalMotor {
            name "motor_2"
          }
        ]
        endPoint Solid {
          translation 6.441142113316787e-05 0.011200011063474371 -0.0005439971226029022
          rotation 0.8702292552334449 -0.34835024731632686 0.3483577881008232 1.709350778561189
          children [
            DEF wheel Shape {
              appearance PBRAppearance {
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.003
                radius 0.01
              }
            }
          ]
          name "solid(1)"
          boundingObject USE wheel
          physics DEF wheel_physics Physics {
            mass 0.05
          }
        }
      }
      HingeJoint {
        jointParameters HingeJointParameters {
          position 5.521652437338447
          axis 0 1 0
          anchor -0.0001 -0.011 -0.00055
        }
        device [
          RotationalMotor {
            name "motor_1"
          }
        ]
        endPoint Solid {
          translation -0.00013590000996114305 -0.010999990907084894 -0.0005439168616232607
          rotation 0.8702292538460614 -0.3483502493934285 0.3483577894895753 1.7093507782549795
          children [
            DEF wheel Shape {
              appearance PBRAppearance {
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.003
                radius 0.01
              }
            }
          ]
          boundingObject USE wheel
          physics DEF wheel_physics Physics {
            mass 0.05
          }
        }
      }
      DistanceSensor {
        translation 0.01 0.005 0
        rotation 0 0 1 0.166
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 1 1 0
              roughness 1
              metalness 0
            }
            geometry Box {
              size 0.001 0.002 0.002
            }
          }
        ]
        name "ds_left_1"
      }
      DistanceSensor {
        translation 0.01 0.005 0
        rotation 0 0 1 0.753
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 1 1 0
              roughness 1
              metalness 0
            }
            geometry Box {
              size 0.001 0.002 0.002
            }
          }
        ]
        name "ds_left_2"
      }
      DistanceSensor {
        translation 0.01 0.005 0
        rotation 0 0 1 1.506
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 1 1 0
              roughness 1
              metalness 0
            }
            geometry Box {
              size 0.001 0.002 0.002
            }
          }
        ]
        name "ds_left_3"
      }
      DistanceSensor {
        translation 0.01 -0.005 0
        rotation 0 0 1 -0.166
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 1 1 0
              roughness 1
              metalness 0
            }
            geometry Box {
              size 0.001 0.002 0.002
            }
          }
        ]
        name "ds_right_1"
      }
      DistanceSensor {
        translation 0.01 -0.005 0
        rotation 0 0 1 -0.753
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 1 1 0
              roughness 1
              metalness 0
            }
            geometry Box {
              size 0.001 0.002 0.002
            }
          }
        ]
        name "ds_right_2"
      }
      DistanceSensor {
        translation 0.01 -0.005 0
        rotation 0 0 1 -1.506
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 1 1 0
              roughness 1
              metalness 0
            }
            geometry Box {
              size 0.001 0.002 0.002
            }
          }
        ]
        name "ds_right_3"
      }
    ]
    boundingObject USE Body
    physics Physics {
      mass 1
      centerOfMass [
        0 0 -0.01
      ]
    }
  }
}